core/domain/stats_domain.py

```diff
@@ -1351,3 +1352,115 @@ def validate(self):
             raise utils.ValidationError(
                 'calculation_output is too big to be stored (size: %d): %s' % (
                     sys.getsizeof(output_data), str(output_data)))
+
+
+class LearnerAnswerDetails(object):
+    """Domain object that represents the answer details submitted by the
+    learner.
+    """
+
+    def __init__(
+            self, state_reference, entity_type, interaction_id,
+            learner_answer_info_list, schema_version=(
+                feconf.CURRENT_LEARNER_ANSWERS_DETAILS_SCHEMA_VERSION)):
+        """Constructs a LearnerAnswerDetail domain object.
+
+            Args:
+                state_reference: str. The reference to the state.
+                entity_type: str. The type of entity, eg. exploration,
+                    question etc.
+                interaction_id: str. The ID of the interaction.
+                learner_answer_info_list: list. The list of LearnerAnswerInfo
+                    objects and version specified in this object.
+                schema_version: str. The schema version of this object.
```

Comment:
What does "this object" refer to?

Why is it a string?

Reply:
Done